test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,352 @@
+		config.DNSServerOverride = dnsServiceIP
+	}
+	err := data.mutateAntreaConfigMap(nil, agentChanges, false, true)
+	require.NoError(t, err, "Error when setting up customDNS server IP in Antrea configmap : %v", err)
+
+	if dnsServiceIP == "" {
+		t.Logf("removing dns server IP from antrea agent as part of teardown")
+	} else {
+		t.Logf("dns server value set to %+v in antrea \n", dnsServiceIP)
+	}
+
+}
+
+func createDnsConfig(t *testing.T, ipv4Address, ipv6Address, domainName string) map[string]string {
+	const coreFileTemplate = `lfx.test:53 {
+        errors
+        log
+        health
+        hosts {
+            {{ if .IPv4 }}{{ .IPv4 }} {{ $.FQDN }}{{ end }}
+            {{ if .IPv6 }}{{ .IPv6 }} {{ $.FQDN }}{{ end }}
+            no_reverse
+            pods verified
+            ttl 10
+        }
+        loop
+        reload
+    }`
+
+	generateConfigData := func() (string, error) {
+		data := struct {
+			IPv4 string
+			IPv6 string
+			FQDN string
+		}{
+			IPv4: ipv4Address,
+			IPv6: ipv6Address,
+			FQDN: domainName,
+		}
+
+		tmpl, err := template.New("configMapData").Parse(coreFileTemplate)
+		if err != nil {
+			return "", err
+		}
+		var output bytes.Buffer
+		err = tmpl.Execute(&output, data)
+		if err != nil {
+			return "", err
+		}
+		return strings.TrimSpace(output.String()), nil
+	}
+
+	configMapData, err := generateConfigData()
+	require.NoError(t, err, "error processing configData template for DNS, ", err)
+	configData := map[string]string{
+		"Corefile": configMapData,
+	}
+
+	return configData
+}
+
+func createDnsPod(t *testing.T, data *TestData) {
+	const (
+		customDnsImage         = "coredns/coredns:1.11.3"
+		customDnsContainerName = "coredns"
+		customDnsVolume        = "config-volume"
+	)
+	volume := []corev1.Volume{
+		{
+			Name: customDnsVolume,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: customDnsConfigName,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  "Corefile",
+							Path: "Corefile",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	volumeMount := []corev1.VolumeMount{
+		{
+			Name:      customDnsVolume,
+			MountPath: "/etc/coredns",
+		},
+	}
+
+	require.NoError(t, NewPodBuilder(customDnsPodName, data.testNamespace, customDnsImage).WithLabels(map[string]string{customDnsLabelKey: customDnsLabelValue}).WithContainerName(customDnsContainerName).WithArgs([]string{"-conf", "/etc/coredns/Corefile"}).AddVolume(volume).AddVolumeMount(volumeMount).WithAnnotations(map[string]string{randomPatchAnnotationKey: randSeq(annotationValueLen)}).Create(data))
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, customDnsPodName, data.testNamespace))
+}
+
+func createFqdnPolicyInNamespace(t *testing.T, data *TestData, domainName string) {
+	const fqdnPolicyName = "test-anp-fqdn"
+	podSelectorLabel := map[string]string{
+		fqdnPodSelectorLabelKey: fqdnPodSelectorLabelValue,
```

Comment:
there is some inconsistency in the function parameters: why is `domainName` a parameter, while `fqdnPodSelectorLabelKey` / `fqdnPodSelectorLabelValue` are accessed as constants. This is why all the constants you still have at the top-level can safely be moved as a function-level constants in the main Test function IMO (see my other comment), and then you can add parameters to `createFqdnPolicyInNamespace` as needed.